### DISCUSSION TITLE
Help for resolving 304 - Not Modified error for our API

### DISCUSSION THREAD

**Discussion by sanjeev-scm** [Sunbird-Lern/Community#162] [Query] [answered]
Hi Team,

Sunbird release version (tags) - release-4.10.0_RC12

We've created a new API for the Sunbird ED portal.
The API is written in python and is not a micro-service.
The API is deployed on the VM.

The API URL for the report with pagination is as follows:
{{host}}/learner/course/v1/progress/reports/do_113876534913794048133?batchid=0138765499798814726&limit=20&offset=0

The same API is used to get the data for exporting the data
{{host}}/learner/course/v1/progress/reports/do_113876534913794048133?batchid=0138765499798814726
where we do not pass the pagination parameters.

We get the data for couple of times with 200 OK status.
Then we get the HTTP 304 - Not Modified status This is related to caching.
We are unable to handle the data in the success handler as the control is going to the error handler. 

As sunbird API's handle different kinds of requests with the same API, 
please let us know any configuration / steps that needs to be done on the API in resolving the 304 - Not modified HTTP STATUS.

Also any suggestions are welcome.

@r0han-9 @Nithin-ns24

**Comment by Jayaprakash8887** [comment]
@sanjeev-scm Not sure if I am understanding the query correctly.

Is the '{{host}}/learner/course/v1/progress/reports/do_113876534913794048133?batchid=0138765499798814726' custom API you have written in Python on which you are receving 304 error response?

**Reply by Jayaprakash8887** [reply]
https://stackoverflow.com/questions/26166433/how-to-prevent-request-that-returns-304 might help you with some insights

**Comment by Jayaprakash8887** [comment]
@maheshkumargangula any inputs?

**Comment by sanjeev-scm** [comment]
> @sanjeev-scm Not sure if I am understanding the query correctly.
> 
> Is the '{{host}}/learner/course/v1/progress/reports/do_113876534913794048133?batchid=0138765499798814726' custom API you have written in Python on which you are receving 304 error response?

Yes, the custom API written in Python on which I am receiving 304 error response

**Reply by AmiableAnil** [reply]
@sanjeev-scm please try these steps https://www.hostinger.in/tutorials/304-status-code

@gandham-santhosh any inputs from devops side?

**Reply by gandham-santhosh** [reply, marked as answer]
@sanjeev-scm can you please try to make the call via API manager and ry ? i.e. {host}/api/{api-endpoint}

**Reply by mohitga** [reply]
@sanjeev-scm,  i am marking the thread as answered, Please feel free to open the thread in case you have any query on the same.
